client/vnsLIf_service.go

```diff
@@ -0,0 +1,98 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/ciscoecosystem/aci-go-client/container"
+	"github.com/ciscoecosystem/aci-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func (sm *ServiceManager) CreateLogicalInterface(name string, l4_l7_devices string, tenant string, nameAlias string, vnsLIfAttr models.LogicalInterfaceAttributes) (*models.LogicalInterface, error) {
```

Comment:
As there are multiple DN possible for this I would use parent_dn to replace l4_l7_devices string, tenant string in this function and all subsequent ones.